src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -332,7 +343,183 @@ public async Task ApproveReviewAsync(ClaimsPrincipal user, string reviewId, stri
+        /// <returns></returns>
+        private async Task MarkAssociatedReviewsForNamespaceReview(List<string> associatedReviewIds, string userId, DateTime requestedOn)
+        {
+            try
+            {
+                var updatedCount = 0;
+
+                foreach (var reviewId in associatedReviewIds.Where(id => !string.IsNullOrEmpty(id)))
+                {
+                    try
+                    {
+                        var review = await _reviewsRepository.GetReviewAsync(reviewId);
+                        if (review != null && LanguageHelper.IsSDKLanguageOrTypeSpec(review.Language))
+                        {
+                            review.NamespaceReviewStatus = NamespaceReviewStatus.Pending;
+                            review.NamespaceApprovalRequestedBy = userId;
+                            review.NamespaceApprovalRequestedOn = requestedOn;
+                            
+                            await _reviewsRepository.UpsertReviewAsync(review);
+                            updatedCount++;
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        _telemetryClient.TrackException(ex);
+                        // Continue with other reviews
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                _telemetryClient.TrackException(ex);
+                // Continue - don't fail the namespace request if this fails
+            }
+        }
+
+        /// <summary>
+        /// Find related language reviews by looking up pull request numbers from TypeSpec review
+        /// </summary>
+        /// <param name="typeSpecReviewId">The TypeSpec review ID</param>
+        /// <returns>List of review IDs for related language reviews</returns>
+        private async Task<List<string>> FindRelatedReviewsByPullRequestAsync(string typeSpecReviewId)
+        {
+            var relatedReviewIds = new List<string>();
+            
+            try
+            {
+                // Get all revisions for the TypeSpec review to find pull request numbers
+                var revisions = await _apiRevisionsManager.GetAPIRevisionsAsync(typeSpecReviewId);
+                var pullRequestNumbers = revisions
+                    .Where(r => r.PullRequestNo.HasValue && r.PullRequestNo.Value > 0)
+                    .Select(r => r.PullRequestNo.Value)
+                    .Distinct()
+                    .ToList();
+
+                // For each pull request number, find related reviews
+                foreach (var prNumber in pullRequestNumbers)
+                {
+                    try
+                    {
+                        // Look up pull requests in azure-rest-api-specs repository (most common for TypeSpec)
+                        var pullRequests = await _pullRequestsRepository.GetPullRequestsAsync(prNumber, "azure-rest-api-specs");
```

Comment:
has this code been specifically tested? is this working? 

I took a look on the database and the name of the repo is "Azure/azure-rest-api-specs`

the query check for the actual value `$"SELECT * FROM PullRequests c WHERE c.PullRequestNumber = {pullRequestNumber} and c.RepoName = '{repoName}' AND c.IsDeleted = false";` not contains or anything, so I doubt this is returning something and my concern is that if we didn't noticed this, has this being validated?